### DISCUSSION TITLE
Remove the default button outline from normalize.css

### DISCUSSION THREAD

**Discussion by justingolden21** [tailwindlabs/tailwindcss#3510] [Ideas]
### Problem

Every time anyone ever writes code about the `:focus` state of something, they need to explicitly mention `button` as well because of this.

Example:

```css
*:focus {
	@apply outline-none ring-2 ring-indigo-500 ring-opacity-50;
}
```

does not work and the TailwindCSS user must specify `button` every time just to overwrite this style:

```diff
-*:focus {
+*focus, button:focus {
	@apply outline-none ring-2 ring-indigo-500 ring-opacity-50;
}
```

### Reasoning

This seems against Tailwind's motto of being "unopinionated" and makes code harder to write.

The CSS above can be vaguely translated into:

```css
:focus,
button:focus {
	outline: 2px solid transparent;
	outline-offset: 2px;
	box-shadow: var(--tw-empty, ) 0 0 0 2px rgba(99, 102, 241, 0.5);
}
```

Which works like so on focused `buttons`:
![image](https://user-images.githubusercontent.com/30274440/106701351-5a0bef80-659b-11eb-8435-51de536549b3.png)

As well as links:
![image](https://user-images.githubusercontent.com/30274440/106701370-64c68480-659b-11eb-9cbc-2a9df9f4900e.png)

However, remove the explicity mention of `button`:

```diff
+*:focus {
-*focus, button:focus {
	@apply outline-none ring-2 ring-indigo-500 ring-opacity-50;
}
```
And we get this in Chrome:
![image](https://user-images.githubusercontent.com/30274440/106701444-81fb5300-659b-11eb-91d9-024775292c88.png)

this in Firefox:
![image](https://user-images.githubusercontent.com/30274440/106701480-95a6b980-659b-11eb-98a2-760f8e3711eb.png)

and this in Edge:
![image](https://user-images.githubusercontent.com/30274440/106701507-a0614e80-659b-11eb-8051-86bd49c5f8df.png)

Yuck.

### Solution

The culprit is this line created inside TailwindCSS:

```css
button:focus {
	outline: 1px dotted;
	outline: 5px auto -webkit-focus-ring-color;
}
```

Removing this line cleans everything back up:

Chrome:
![image](https://user-images.githubusercontent.com/30274440/106701601-c5ee5800-659b-11eb-917b-72e36228443d.png)

Firefox:
![image](https://user-images.githubusercontent.com/30274440/106701642-d6063780-659b-11eb-9ec3-5541874a73ed.png)

Edge:
![image](https://user-images.githubusercontent.com/30274440/106701663-def70900-659b-11eb-8c05-e573af9c4198.png)

### Counterargument Rebuttal

I understand that the default outline is there to make sure buttons are shown an outline if no styles are provided.

However, this impedes on the user writing their own styles, and creates inconsistency within the user-interface, which are two things I believe Tailwind is very much opposed to.

**Comment by justingolden21** [comment]
By the way, this is off topic, but it seems that in `var(--tw-empty, )` both the comma and the space after the comma must be there or the styles break; does anybody know why this behaves like this?

**Reply by adamwathan** [reply]
This is intentional and necessary, the empty space is valid content. Read towards the end of this blog post for a better understanding of how it works:

https://adamwathan.me/composing-the-uncomposable-with-css-variables/

**Comment by adamwathan** [comment]
Hey! No plans to fork normalize and change it I'm afraid, so I would recommend just authoring your global focus styles the way you demonstrated:

```css
*:focus, button:focus {
	@apply outline-none ring-2 ring-indigo-500 ring-opacity-50;
}
```

Not really an annoying or high effort solution.
